packages/number-field/src/NumberField.ts

```diff
@@ -311,24 +311,24 @@ export class NumberField extends TextfieldBase {
     protected onFocus(): void {
         super.onFocus();
         this._trackingValue = this.inputValue;
-        this.keyboardFocused = true;
+        this.keyboardFocused = !this.readonly && true;
```

Comment:
Are these changes just a fix you're sticking in there? Or are they relevant to the rest of the work? Just curious :-)

Reply:
This comes from a change in the `readonly` delivery in Spectrum. Text field centric UIs will no longer deliver the text field background when readonly and this prevents them from delivering the focus state in that context.